network/nattype.go

```diff
@@ -0,0 +1,26 @@
+package network
+
+// NATDeviceType indicates the type of the NAT device i.e. whether it is a Hard or an Easy NAT.
+type NATDeviceType int
+
+const (
+	// NATDeviceTypeUnknown indicates that the type of the NAT device is unknown.
+	NATDeviceTypeUnknown NATDeviceType = iota
+
+	// NATDeviceTypeEasy indicates that the NAT device is an Easy NAT i.e. it supports consistent endpoint translation.
```

Comment:
Can we provide the precise definitions of
(a) the heuristic used to determine this
and
(b) what we mean by 'consistent endpoint translation' - do you mean that the externally bound host:port is the same for all remote hosts, stable for different ports of a single remote host, or once mapped allows inbound from a specifically negotiated remote host?

Reply:
I've updated the docs to use the terms used in scientific literature and explain consistent endpoint translation in relation to those terms.